### PR TITLE
Added drag/drop functionality to complaint and complaint details page.

### PR DESCRIPTION
**Pull request by tanvirp10**
Added Drag/drop functionality code and javascript to "Add.cshtml" and "Details.cshtml". Added server-sided logic to include message "Files uploaded successfully" on Compaint Details page in file "Details.cshtml.cs". Also added CSS styling to include hover screen for complaint details page in "Details.cshtml" file.

### CODE REVIEW

**Review by dougwaldron — CHANGES_REQUESTED** [review]
Tanvir, this is looking great! I've wanted this feature for a long time, and I really like the look and feel of the UI you created.

Here are some next steps to work on:

* Implement the drop target styling on the Add Complaint page like the Complaint Details page.

* In the "Details.cshtml.cs" file, add a `[TempData]` attribute to the `HighlightId` property. This is an ASP.NET feature that lets you save data for use after an internal Redirect. Meaning, you can directly set `HighlightId = true` in the "Post" method and remove the changes you made to the "Get" method. The property will then already be set when the Details page reloads.

* On the Details page, the drop target has a fixed height. When adding multiple files, the content overflows the container. This is working correctly on the Add Complaint page. Consider listing all the files in a single line or sentence rather than one per line.

* The JavaScript should be moved to an external script that can be called by both pages (DRY). The only difference between them is clearing the overall success message which can handled by checking first to see if a message element exists.

* The JS can be made more generic by attaching to elements based on class rather than ID. This makes the code more maintainable long-term and would also allow multiple drop targets to be defined per page. As part of this change, don't use an `onclick` attribute on the button, but add a click event listener from the JS instead.

* Move the custom CSS to the "site.css" file so it's available to both pages. When adding CSS, it's very important to ensure future maintainability. This means you should use existing classes or CSS variables wherever possible, so that it's better able to handle future theme updates. And if you have to add new code, make it as reusable and generic as possible, i.e., minimize specificity, don't use inline style attributes, etc. For example:

    - Bootstrap has built-in [display](https://getbootstrap.com/docs/5.3/utilities/display/) and [flex](https://getbootstrap.com/docs/5.3/utilities/flex/) classes that can be used without requiring any new CSS.

    - Bootstrap also has lots of [border utilities](https://getbootstrap.com/docs/5.3/utilities/borders/) that can set border width and color. What's missing is the dashed border style, but instead of adding a dashed border specifically to the drop-area, create a reusable `.border-dashed` utility.

Let me know if you have any questions about any of these points and keep up the great work!

**Review by tanvirp10 — COMMENTED** [review]
Implemented similar styling on Add and Details page.
Added property and attribute to Details.cshtml.cs page.
Moved Javascript from pages to a separate page.
Updated Javascript to include event listener attributes.
Moved the custom CSS to site.css
Fixed contained overflowing issue when multiple files selected.

File: src/WebApp/wwwroot/css/site.css [hunk 1]
```diff
@@ -417,3 +417,64 @@ svg:not(:root) {
...
+    content: "Drop files here";
+    position: absolute;
+    top: 50%;
+    left: 50%;
+    transform: translate(-50%, -50%);
+    color: #007bff;
+    font-size: 1.5em;
+    text-align: center;
+    z-index: 10;
+}
```
Comment (dougwaldron) [inline src/WebApp/wwwroot/css/site.css:454]: These can be removed:

```css
.drop-area.hover::before {
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    text-align: center;
}
```

File: src/WebApp/wwwroot/css/site.css [hunk 2]
```diff
@@ -417,3 +417,64 @@ svg:not(:root) {
...
+    content: "";
+    position: absolute;
+    top: 0;
+    left: 0;
+    width: 100%;
+    height: 100%;
+    background-color: rgba(0, 123, 255, 0.1);
+    z-index: 5;
+    backdrop-filter: blur(5px);
+}
```
Comment (dougwaldron) [inline src/WebApp/wwwroot/css/site.css:466]: These can be removed:

```css
.drop-area.hover::after {
    top: 0;
    left: 0;
    z-index: 5;
}
```

File: src/WebApp/wwwroot/css/site.css [hunk 3]
```diff
@@ -417,3 +417,64 @@ svg:not(:root) {
...
+
+.drop-area p {
+    margin: 0 0 10px 0;
+    text-align: center;
```
Comment (dougwaldron) [inline src/WebApp/wwwroot/css/site.css:470]: `text-align: center;` here can be removed.

**Review by dougwaldron — CHANGES_REQUESTED** [review]
I think these edits will finish this up.

**Review by tanvirp10 — COMMENTED** [review]
Added original paragraphs back to Add.cshtml & Details.cshtml.
Added bootstrap elements to Add.cshtml & Details.cshtml.
Added TempData attribute and removed uploadSuccess bool from Details.cshtml.cs.
Cleaned up & simplified CSS styling for drag & drop and hover feature.
Cleaned up fileUpload.js Javascript code.

**Review by dougwaldron — APPROVED** [review]

**Pull request merged**